packages/widget/src/controllers/transfers/fungible-token-transfer.ts

```diff
@@ -247,6 +296,11 @@ export class FungibleTokenTransferController implements ReactiveController {
         await getRoutes(this.env, sourceNetwork.chainId, 'fungible')
       );
     }
+
+    if (!this.substrateProvider && sourceNetwork.type === Network.SUBSTRATE) {
```

Comment:
why do we need this here?
Does it make more sense to not keep `this.substrateProvider` instead we replace it with getSubstrateProvider which returns provider based on selected source network?